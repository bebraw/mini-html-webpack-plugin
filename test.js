const MiniHtmlWebpackPlugin = require('./index');
const compiler = require('@webpack-contrib/test-utils');

const getConfig = (options, config = {}) =>
	Object.assign(
		{
			entry: ['./index.js'],
			plugins: [new MiniHtmlWebpackPlugin(options)],
		},
		config
	);

test('default options', () => {
	return compiler({}, getConfig()).then(result => {
		expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
	});
});

test('custom title', () => {
	return compiler({}, getConfig({ context: { title: 'Pizza' } })).then(
		result => {
			expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
		}
	);
});

test('custom lang', () => {
	return compiler(
		{},
		getConfig({ context: { htmlAttributes: { lang: 'ru' } } })
	).then(result => {
		expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
	});
});

test('custom js attribute', () => {
	return compiler(
		{},
		getConfig({ context: { jsAttributes: { defer: 'defer' } } })
	).then(result => {
		expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
	});
});

test('custom template', () => {
	return compiler(
		{},
		getConfig({
			context: { title: 'Pizza', htmlAttributes: { lang: 'it' } },
			template: ({ title }) => `<div>${title}</div>`,
		})
	).then(result => {
		expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
	});
});

test('custom filename', () => {
	const filename = 'pizza.html';
	return compiler({}, getConfig({ filename })).then(result => {
		expect(result.compilation.assets[filename]._value).toMatchSnapshot();
	});
});

test('custom publicPath', () => {
	return compiler({}, getConfig({ publicPath: 'pizza/' })).then(result => {
		expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
	});
});
