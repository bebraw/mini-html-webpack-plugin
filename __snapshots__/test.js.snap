// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom filename 1`] = `
"<!DOCTYPE html>
  <html lang=\\"en\\">
    <head>
      <meta charset=\\"UTF-8\\">
      <title></title>
      
    </head>
    <body>
      <script src=\\"runtime~main.js\\"></script><script src=\\"main.js\\"></script>
    </body>
  </html>"
`;

exports[`custom js attribute 1`] = `
"<!DOCTYPE html>
  <html lang=\\"en\\">
    <head>
      <meta charset=\\"UTF-8\\">
      <title></title>
      
    </head>
    <body>
      <script src=\\"runtime~main.js\\" defer=\\"defer\\"></script><script src=\\"main.js\\" defer=\\"defer\\"></script>
    </body>
  </html>"
`;

exports[`custom lang 1`] = `
"<!DOCTYPE html>
  <html lang=\\"ru\\">
    <head>
      <meta charset=\\"UTF-8\\">
      <title></title>
      
    </head>
    <body>
      <script src=\\"runtime~main.js\\"></script><script src=\\"main.js\\"></script>
    </body>
  </html>"
`;

exports[`custom publicPath 1`] = `
"<!DOCTYPE html>
  <html lang=\\"en\\">
    <head>
      <meta charset=\\"UTF-8\\">
      <title></title>
      
    </head>
    <body>
      <script src=\\"pizza/runtime~main.js\\"></script><script src=\\"pizza/main.js\\"></script>
    </body>
  </html>"
`;

exports[`custom template 1`] = `"<div>Pizza</div>"`;

exports[`custom title 1`] = `
"<!DOCTYPE html>
  <html lang=\\"en\\">
    <head>
      <meta charset=\\"UTF-8\\">
      <title>Pizza</title>
      
    </head>
    <body>
      <script src=\\"runtime~main.js\\"></script><script src=\\"main.js\\"></script>
    </body>
  </html>"
`;

exports[`default options 1`] = `
"<!DOCTYPE html>
  <html lang=\\"en\\">
    <head>
      <meta charset=\\"UTF-8\\">
      <title></title>
      
    </head>
    <body>
      <script src=\\"runtime~main.js\\"></script><script src=\\"main.js\\"></script>
    </body>
  </html>"
`;
